### PR TITLE
DM-8542 Add support for changing draw layer symbol and the size

### PR DESCRIPTION
**Pull request by cwang2016**
The development mainly adds support for the user to change the symbol and its size when overlaying a catalog on an image including the features: 
- catalog default symbol is square and size 3 to 20, size means the size shown on UI. 
- add shape options for the symbol, 
   x, square, cross, diamond, circle, arrow, boxcircle, and dot. 
- create a dialog for changing the symbol and its size, each drawing layer has its own dialog. 
- the change in color will update symbol content in symbol dialog and drawing layer panel in real time. 
- the change in symbol, size, and color will update the catalog symbol overlaying the image in real time.

### CODE REVIEW

File: src/firefly/js/ui/PointShapeSizePicker.jsx [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+    "navajowhite":"#ffdead","navy":"#000080",
+    "oldlace":"#fdf5e6","olive":"#808000","olivedrab":"#6b8e23","orange":"#ffa500","orangered":"#ff4500","orchid":"#da70d6",
+    "palegoldenrod":"#eee8aa","palegreen":"#98fb98","paleturquoise":"#afeeee","palevioletred":"#d87093","papayawhip":"#ffefd5","peachpuff":"#ffdab9","peru":"#cd853f","pink":"#ffc0cb","plum":"#dda0dd","powderblue":"#b0e0e6","purple":"#800080",
+    "rebeccapurple":"#663399","red":"#ff0000","rosybrown":"#bc8f8f","royalblue":"#4169e1",
+    "saddlebrown":"#8b4513","salmon":"#fa8072","sandybrown":"#f4a460","seagreen":"#2e8b57","seashell":"#fff5ee","sienna":"#a0522d","silver":"#c0c0c0","skyblue":"#87ceeb","slateblue":"#6a5acd","slategray":"#708090","snow":"#fffafa","springgreen":"#00ff7f","steelblue":"#4682b4",
+    "tan":"#d2b48c","teal":"#008080","thistle":"#d8bfd8","tomato":"#ff6347","turquoise":"#40e0d0",
+    "violet":"#ee82ee",
+    "wheat":"#f5deb3","white":"#ffffff","whitesmoke":"#f5f5f5",
+    "yellow":"#ffff00","yellowgreen":"#9acd32"};
+
```
Comment (robyww) [inline src/firefly/js/ui/PointShapeSizePicker.jsx]: are this colors for you background algorithm? You might want to clean it up so it uses single quotes and the keys are not quoted.

Comment (robyww) [inline src/firefly/js/ui/PointShapeSizePicker.jsx]: I understand more this is a named color map to RGB. I think you want to this objects and the background utility routines to util/Color.js.

File: src/firefly/js/ui/PointShapeSizePicker.jsx [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+            hsv.saturation /= 100.0;
+            hsv.value /= 100.0;
+
+
+            let i = Math.floor(hsv.hue);
+            let f = hsv.hue - i;
+            let a = hsv.value * (1.0 - hsv.saturation);
+            let b = hsv.value * (1.0 - hsv.saturation * f);
+            let c = hsv.value * (1.0 - hsv.saturation * (1-f));
+            let d = hsv.value;
```
Comment (robyww) [inline src/firefly/js/ui/PointShapeSizePicker.jsx]: you could use `const` for these instead of `let`

**Comment by ejoliet** [comment]
Yay!! Great extra feature for catalogs use on images. I think the options are clear enough and the fact that the change is applied immediately is also a extra bonus for usability. I like it a lot. Good job!

**Comment by cwang2016** [comment]
Some update:
- only one symbol dialog is rendered on the selected symbol on the draw layer panel.
- move  color related routines to Color.js

**Pull request merged**